sdk/core/azure-core-test/src/main/java/com/azure/core/test/http/MockHttpResponse.java

```diff
@@ -182,7 +182,7 @@ public HttpHeaders getHeaders() {
 
         return bodyBytes == null
                 ? Mono.empty()
-                : Mono.just(new String(bodyBytes, charset));
+                : Mono.just(CoreUtils.bomAwareToString(bodyBytes, getHeaderValue("Content-Type")));
```

Comment:
You'll want to move this into the `getBodyAsString` overload without parameters, nothing I've found in Azure Core leverages the `Charset` overload.

Reply:
```java
public Mono<String> getBodyAsString() {
  return (bodyBytes == null)
    ? Mono.empty()
    : Mono.just(CoreUtils.bomAwareToString(bodyBytes, getHeaderValue("Content-Type"));
}